server/src/main/java/org/elasticsearch/script/IngestScript.java

```diff
@@ -33,18 +33,31 @@ public abstract class IngestScript {
     /** The generic runtime parameters for the script. */
     private final Map<String, Object> params;
 
-    public IngestScript(Map<String, Object> params) {
+    private final Metadata metadata;
+
+    public IngestScript(Map<String, Object> params, Metadata metadata) {
         this.params = params;
+        this.metadata = metadata;
     }
 
     /** Return the parameters for this script. */
     public Map<String, Object> getParams() {
         return params;
     }
 
-    public abstract void execute(Map<String, Object> ctx);
+    /** Provides backwards compatibility access to ctx */
+    public Map<String, Object> getCtx() {
+        return metadata;
```

Comment:
Can we pass ctx separately to ingest script, so that Metadata can remain strongly typed?

Reply:
Yup, done.